### PR TITLE
Updated user guide for 1.0 release

### PR DESCRIPTION
**Pull request by KevinCLydon**
Updated the user guide to reflect the new ways to generate reports, R support in reports, and some updates I missed related to functionality added in previous releases.

### CODE REVIEW

File: UserGuide.md [hunk 1]
```diff
@@ -40,17 +40,20 @@
         * [Concise Example](#concise-example)
         * [Debugging Failed Runs / FAQ](#debugging-failed-runs-faq)
             * [Interrogating Cromwell](#interrogating-cromwell)
-            * [Changing a WDL didn’t change the error message from CARROT / Cromwell](#changing-a-wdl-didn’t-change-the-error-message-from-carrot-cromwell)
+            * [Failures in CARROT](#failures-in-carrot)
     * [Defining and Generating Reports](#defining-and-generating-reports)
         * [Report Definition Process](#report-definition-process)
             * [1. Prepare your Jupyter notebook](#1-prepare-your-jupyter-notebook)
-                * [1a. Python 3](#1a-python-3)
+                * [1a. Python 3 and R 4](#1a-python-3-and-r-4)
```
Comment (jamesemery) [inline UserGuide.md:47]: I almost got up in arms about a "python 4"...

File: UserGuide.md [hunk 2]
```diff
@@ -535,7 +549,22 @@ The optional `--delete_failed `flag will overwrite an existing failed run_report
...
+
+```
+carrot_cli pipeline create_report_for_runs [options] PIPELINE REPORT
+```
+
+`PIPELINE` is the ID or name of the pipeline to which the runs you wish to query belong.
+
+`REPORT` is the ID or name of the report you wish to generate for the run.  If you just completed step 2 above and created a report, the ID of that report should have been included in the results from that command.  If not, you can find the report using `carrot_cli report find`.
+
+`[options]` can be any from a list of options for filtering runs that are the same as you have available to you for any of the `find_runs` commands. For a full list of the filtering options available, see the output of `carrot_cli pipeline create_report_for_runs --help`.
```
Comment (jamesemery) [inline UserGuide.md:567]: I would add more detail here about what options are likely to be sueful... If i don't provide any options at all do i just get the outputs of every run ever completed for PIPELINE by default?

File: UserGuide.md [hunk 3]
```diff
@@ -506,17 +517,20 @@ The `--created_by email` flag is optional and defaults to the email address in y
...
 
-
 `REPORT` is the ID or name of the report you wish to map.  If you just completed step 2 and created a report, the ID of that report should have been included in the results from that command.  If not, you can find the report using `carrot_cli report find`.
 
 `REPORT_TRIGGER` is an optional argument that corresponds to the event that should trigger the generation of the report.  It defaults to `single`, which means a report will be generated upon successful completion of a single run of a test based on the template.  The other option is `pr`, which means a report will be generated upon successful completion of a [PR Comparison run](#starting-a-pr-comparison-run-from-github).
 
 
+#### <a name="generating-a-report"/> **Generating a Report**
 
+Filled reports can be generated in three ways. Once the report has been generated, it will be available in 3 forms: the Jupyter notebook that is produced by CARROT before being run, the filled Jupyter notebook that is created once all the cells are run, and a static HTML representation of the filled notebook.
```
Comment (jamesemery) [inline UserGuide.md:527]: comment on where those get populated to

File: UserGuide.md [hunk 4]
```diff
@@ -506,17 +517,20 @@ The `--created_by email` flag is optional and defaults to the email address in y
...
 
-#### <a name="generating-a-report"/> **Generating a Report**
+##### <a name="auto-generated-reports"/> **Auto-Generated Reports**
 
-Filled reports can be generated in two ways. First, if a report has been properly defined and mapped to a template, any future successful runs of tests based on that template will automatically result in the report being generated, and any users subscribed to the pipeline, template, or test, or who created the run, will received an email notification when the report has been generated.  Additionally, if the run is triggered via a GitHub comment, the results of the report run will also be posted as a comment below it.
+First, if a report has been properly defined and mapped to a template, any future successful runs of tests based on that template will automatically result in the report being generated, and any users subscribed to the pipeline, template, or test, or who created the run, will receive an email notification when the report has been generated.  Additionally, if the run is triggered via a GitHub comment, the results of the report run will also be posted as a comment below it.
```
Comment (jamesemery) [inline UserGuide.md:531]: This doesn't really clarify what happens if there is a report for several runs that matters here? Presumably it does distinguish between multi-reports and single reports that are mapped or does it auto-generate the multi-reports everytime a new run in a PIPELINE is run?

File: UserGuide.md [hunk 5]
```diff
@@ -373,14 +387,14 @@ carrot_cli result create --name "TSV" --description "Generic TSV file." --result
...
+carrot_cli template create --pipeline "HaplotypeCaller DRAGEN Functional Equivalence"  --name "HaplotypeCaller DRAGEN Single Sample Comparison" --description "Comparing the results of HaplotypeCaller and DRAGEN for a single sample." --test_wdl https://storage.googleapis.com/dsp-methods-carrot-data/test_data/Dragen_GATK_Concordance/001_HC_Dragen_test.wdl --eval_wdl https://storage.googleapis.com/dsp-methods-carrot-data/test_data/Dragen_GATK_Concordance/002_HC_Dragen_eval.wdl
 
 # Map the template outputs to CARROT result objects:
-carrot_cli template  map_to_result 50d90287-bf31-46f8-888a-508567b0378d a7924abc-8327-4ae9-901a-f1a42e8447e8 HC_Dragen_eval.out_snp_plot
+carrot_cli template  map_to_result "HaplotypeCaller DRAGEN Single Sample Comparison" a7924abc-8327-4ae9-901a-f1a42e8447e8 HC_Dragen_eval.out_snp_plot
```
Comment (jamesemery) [inline UserGuide.md:393]: so much nicer

File: UserGuide.md (hunk 1 above)
Comment (lbergelson) [reply UserGuide.md:47]: You haven't switched to Python 4 yet?

File: UserGuide.md [hunk 6]
```diff
@@ -77,7 +80,7 @@ v0.6.2
 
 ## <a name="about"/> **About**
 
-This document serves as a guide for how to interact with CARROT, including how to set up tests, link your github repo to CARROT, and interact with the server.  This guide does not cover how to set up the CARROT system - that topic is covered in another document.
+This document serves as a guide for how to interact with CARROT, including how to set up tests, link your github repo to CARROT, and interact with the server.  This guide does not cover how to set up the CARROT system - that topic is explained in the README.
 
 Remember - “A full eval is just a #carrot() away.”
```
Comment (lbergelson) [inline UserGuide.md:85]: Top notch slogan.

File: UserGuide.md [hunk 7]
```diff
@@ -77,7 +80,7 @@ v0.6.2
 
 ## <a name="about"/> **About**
 
-This document serves as a guide for how to interact with CARROT, including how to set up tests, link your github repo to CARROT, and interact with the server.  This guide does not cover how to set up the CARROT system - that topic is covered in another document.
+This document serves as a guide for how to interact with CARROT, including how to set up tests, link your github repo to CARROT, and interact with the server.  This guide does not cover how to set up the CARROT system - that topic is explained in the README.
```
Comment (lbergelson) [inline UserGuide.md:83]: I'd put a link to the readme there.

File: UserGuide.md [hunk 8]
```diff
@@ -143,7 +146,7 @@ The relationship between **pipelines, templates, tests, runs**, and **results**
 
 ![Test Structure](images/Test_Structure.png)
 
-The following sections detail some specifics for how to use CARROT to create and dispatch analyses.  These sections refer to “tests” largely as notional concepts, rather than the CARROT **test** defined above.  In the future the terminology will be updated to be less ambiguous / confusing.
+The following sections detail some specifics for how to use CARROT to create and dispatch analyses.  These sections refer to “tests” largely as notional concepts, rather than the CARROT **test** defined above.
```
Comment (lbergelson) [inline UserGuide.md:149]: Kind of confusing to define test and then say "well, we're going to ignore that."  I'm not sure "notional concepts" is what you mean.  (Unless you mean tests are imaginary concepts.)

File: UserGuide.md [hunk 9]
```diff
@@ -154,23 +157,25 @@ The best way to create tests for CARROT is to start with a WDL that already exis
...
 
 
 ##### <a name="mapping-test-outputs-to-eval-inputs"/> **Mapping Test Outputs to Eval Inputs**
 
-It is encouraged to map outputs from your test WDL to inputs for your evaluation WDL.  To do so requires using a specific format for the value of the input supplied to your evaluation WDL.  For example, if your test WDL defines a workflow named “Test_Workflow”, which has an output called “output_vcf”, and you want to supply that as an input called “test_vcf” to a workflow called “Eval_Workflow” defined in your evaluation WDL, your input json for the evaluation WDL should contain the following:
+In order to evaluate the outputs of your test WDL in your evaluation WDL, you'll need to map those outputs to the evaluation WDL's inputs.  To do so requires using a specific format for the value of the input supplied to your evaluation WDL.  For example, if your test WDL defines a workflow named “Test_Workflow”, which has an output called “output_vcf”, and you want to supply that as an input called “test_vcf” to a workflow called “Eval_Workflow” defined in your evaluation WDL, your input json for the evaluation WDL should contain the following:
```
Comment (lbergelson) [inline UserGuide.md:178]: Is there any sort of checker/linter that wdl writers can use to validate that these line up?

File: UserGuide.md [hunk 10]
```diff
@@ -181,9 +186,15 @@ It is encouraged to map outputs from your test WDL to inputs for your evaluation
...
+
+In order to use a custom docker image, first a **[software](#carrot-terms-and-structure)** record must be created in CARROT, either using the REST API or carrot_cli, that references the git repository.  Once that is done, the custom build can be specified as an input to either your test WDL or your evaluation WDL.  For example, if you have created an entry for a software called “<code><em>my_cool_software</em></code>” and you want to generate a docker image from the commit with the hash “<code><em>ca82a6dff817ec66f44342007202690a93763949</em></code>” to supply to an input named “<code><em>docker_image</em></code>” in a workflow titled “<code><em>Test_Workflow</em></code>” in your test WDL, your test input json should contain the following:
+
+`"Test_Workflow.docker_image":"image_build:my_cool_software|ca82a6dff817ec66f44342007202690a93763949"`
```
Comment (lbergelson) [inline UserGuide.md:193]: Do you need the full commit hash or is the short hash ok?

File: UserGuide.md [hunk 11]
```diff
@@ -181,9 +186,15 @@ It is encouraged to map outputs from your test WDL to inputs for your evaluation
...
+
+It is also possible to specify the commit to build from using tags.  For example, if “<code><em>my_cool_software</em></code>” has a commit tagged “<code><em>0.5.9</em></code>”, you can build from that commit like this:
 
-`"Test_Workflow.docker_image":"image_build:my_cool_software|ca82a6dff817ec66f44342007202690a93763949`”
+`"Test_Workflow.docker_image":"image_build:my_cool_software|0.5.9"`
```
Comment (lbergelson) [inline UserGuide.md:197]: What about branch names?  Can it take anything that is a valid git reference  (HEAD^) or is it only tags/hashes?

File: UserGuide.md [hunk 12]
```diff
@@ -308,6 +319,8 @@ The  `TEMPLATE ID OR NAME` corresponds to the CARROT ID or name of the template
 
 `<EVAL_WDL_DEFAULT_JSON_FILE> `corresponds to the JSON file containing default values for the evaluation WDL.  In general, all workflow inputs should be set here except for the docker container specification (this is required for the GitHub integration to function properly).  If the evaluation WDL does not require a docker image input, you do not need to add a dummy value to the WDL prior to running the test.
 
+The `--test_option_defaults` and `--eval_option_defaults` inputs allow you to optionally specify configuration options (the `workflowOptions` param when [submitting a workflow to Cromwell](https://cromwell.readthedocs.io/en/stable/api/RESTAPI/#submit-a-workflow-for-execution)) for running the test and eval WDLs.  Like the input defaults, this can be overrided or added to when running the test.
```
Comment (lbergelson) [inline UserGuide.md:322]: optionally specify options :)

Comment (lbergelson) [reply UserGuide.md:322]: overrided -> overidden ?

File: UserGuide.md [hunk 13]
```diff
@@ -317,17 +330,18 @@ Now that the test is set up, you can run the test from the command line or[ trig
...
 
 
 The` --name "<RUN NAME>"` flag is optional and will be populated by CARROT if a name is not supplied.
 
 `<TEST_WDL_INPUT_JSON_FILE> `corresponds to the JSON file containing run-specific values for the test WDL.  Often the only thing required will be the docker container specification, though this is test-specific (this is required for the GitHub integration to function properly).
 
-`<EVAL_WDL_INPUT_JSON_FILE> `corresponds to the JSON file containing run-specific values for the  evaluation WDL.
+`<EVAL_WDL_INPUT_JSON_FILE> `corresponds to the JSON file containing run-specific values for the evaluation WDL.
+
+The `--test_options` and `--eval_options` inputs allow you to optionally specify run-specific configuration options (the `workflowOptions` param when [submitting a workflow to Cromwell](https://cromwell.readthedocs.io/en/stable/api/RESTAPI/#submit-a-workflow-for-execution)) for running the test and eval WDLs.
```
Comment (lbergelson) [inline UserGuide.md:344]: I think this phrasing is better than the one above.

File: UserGuide.md [hunk 14]
```diff
@@ -404,22 +418,19 @@ carrot_cli test run --test_input 001_HC_Dragen_test.WGS4.json --eval_input 002_H
...
 
 ##### <a name="interrogating-cromwell"/> **Interrogating Cromwell**
 
-When a run in CARROT fails, often it has proceeded far enough to have a Cromwell run ID associated with it.  If this is the case, you can look directly at the Cromwell failure by interrogating the Cromwell server using the Swagger Web API or by using a tool like [cromshell](https://github.com/broadinstitute/cromshell).  If you choose to use cromshell you must ensure that the cromshell server configuration file in `~/.cromshell/cromwell_server.config` points to the cromwell server that CARROT is using (including the correct port on which to communicate).
+If the failure happened in Cromwell, you can look directly at the Cromwell failure by interrogating the Cromwell server using the Swagger Web API or by using a tool like [cromshell](https://github.com/broadinstitute/cromshell).  If you choose to use cromshell you must ensure that the cromshell server configuration file in `~/.cromshell/cromwell_server.config` points to the cromwell server that CARROT is using (including the correct port on which to communicate).
```
Comment (lbergelson) [inline UserGuide.md:425]: If there's an easy way to print the server and port from the carrot_cli I'd include that command here.

**Review by lbergelson — COMMENTED** [review]
The doc seems useful.  I made some comments, mostly text nitpicks.

File: UserGuide.md (hunk 1 above)
Comment (KevinCLydon) [reply UserGuide.md:47]: Python 4 is strongly-typed and comes with a package manager that is actually good and works.

File: UserGuide.md (hunk 3 above)
Comment (KevinCLydon) [reply UserGuide.md:527]: Oh that's a good idea.  I should tell people where to get those

File: UserGuide.md (hunk 4 above)
Comment (KevinCLydon) [reply UserGuide.md:531]: There's a section a couple before this one that explains the command for mapping a report to a template to generate runs automatically and it explains how to set up the mapping for different types of reports/runs.  I'll add a link to that here with a note.

File: UserGuide.md (hunk 6 above)
Comment (KevinCLydon) [reply UserGuide.md:85]: Lol I can't take credit for it. I can't remember for sure but I wanna say David or Jonn came up with that, maybe?

File: UserGuide.md (hunk 8 above)
Comment (KevinCLydon) [reply UserGuide.md:149]: I'll try to clarify.  Admittedly having an entity named "test" makes things slightly confusing.

File: UserGuide.md (hunk 9 above)
Comment (KevinCLydon) [reply UserGuide.md:178]: There isn't, but that's a good idea.  I'll create an issue for that.

File: UserGuide.md (hunk 10 above)
Comment (KevinCLydon) [reply UserGuide.md:193]: Short hashes should work, but not in a way that I've totally accounted for so I should fix that

File: UserGuide.md (hunk 12 above)
Comment (KevinCLydon) [reply UserGuide.md:322]: I knew those sounded awkward when I wrote them 😵‍💫  I'll fix those

File: UserGuide.md (hunk 14 above)
Comment (KevinCLydon) [reply UserGuide.md:425]: Ooo good idea

Comment (KevinCLydon) [reply UserGuide.md:425]: Oh wait no, there isn't an easy way to do that

Comment (KevinCLydon) [reply UserGuide.md:425]: I should add something like that

File: UserGuide.md (hunk 6 above)
Comment (jonn-smith) [reply UserGuide.md:85]: I have no memory of creating this... I feel like @KevinCLydon did, but ¯\_(ツ)_/¯

**Review by jonn-smith — APPROVED** [review]
Approved as per our conversation.

**Pull request merged**